src/docs.js

```diff
@@ -106,16 +109,26 @@ async function documenter(options) {
   if (options.publish) {
     let creds = options.aws;
     if (!creds) {
-      let auth = new client.Auth({
-        credentials: options.credentials,
-        baseUrl: options.authBaseUrl,
-      });
-
-      creds = await auth.awsS3Credentials('read-write', options.bucket, options.project + '/');
+      //Bypassing Tascluster credentials if none are provided in the config file.
```

Comment:
I think the easiest way to fix this is to just pass `creds` to the documenter; then the conditional on line 111 (`if (!creds)`) will end up skipping all of this.

Reply:
@djmitche I'm a bit confused by this comment... The way I'm understanding it, the `creds` variable is set to `options.aws` which is null by default in the docs.js file.

https://github.com/taskcluster/taskcluster-lib-docs/blob/master/src/docs.js#L107

So even if I pass creds to the documenter function in the tests, the `creds` variable will always be empty and so the `auth.awsS3Credentials` call always ends up executing... Not sure how to bypass that in a different way (though I'm sure there must be a better solution...), maybe by changing the `creds` assignement from `options.aws` to something else? Otherwise I think I've done all the other requested changes :)

Reply:
Yes, I meant to do what's necessary so that `creds` is set, which would be to pass `aws` in the options to `documenter`.  For example, in
https://github.com/taskcluster/taskcluster-lib-docs/blob/8b235ff18325335b8075cc0db3cf1a2b610e4ea9/test/main_test.js#L118-L133

I would remove `credentials` and pass a fake value for `aws` (it doesn't matter what value, since your mock `S3UploadStream` doesn't actually care what `creds` is set to).